server/lib/src/plugins/ecdhkeygen.rs

```diff
@@ -0,0 +1,142 @@
+use openssl::ec::{EcGroup, EcKey};
+use openssl::nid::Nid;
+use sketching::admin_error;
+
+use super::Plugin;
+use crate::event::CreateEvent;
+use crate::prelude::{Entry, EntryInvalidNew, OperationError};
+use crate::server::QueryServerWriteTransaction;
+use crate::value::PartialValue;
+use sketching::tagged_event;
+use sketching::EventTag;
+
+lazy_static! {
+    // it contains all the partialvalues used to match against an Entry's class,
+    // we just need a partialvalue to match in order to target the entry
+    static ref CLASSES_TO_UPDATE: [PartialValue; 1] = [PartialValue::new_iutf8("account")];
+    static ref DEFAULT_KEY_GROUP: EcGroup = {
+        let nid = Nid::X9_62_PRIME256V1; // NIST P-256 curve
+        EcGroup::from_curve_name(nid).unwrap()
+    };
+}
+pub struct EcdhKeyGen {}
+
+impl EcdhKeyGen {
+    fn is_entry_to_update<VALUE, STATE>(entry: &mut Entry<VALUE, STATE>) -> bool {
+        CLASSES_TO_UPDATE
+            .iter()
+            .any(|pv| entry.attribute_equality("class", pv))
+    }
```

Comment:
You know what I'm about to say here ... :)

Reply:
yep ..